### ISSUE TITLE
Changing Variable to a JSON value in the UI makes next Terraform run fail

### ISSUE THREAD

**Issue by felixpelletier** [PrefectHQ/terraform-provider-prefect#254]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform Version

terraform 0.15.3 (I know it's old, but I don't believe the issue is there)
provider v2.2.2

### Affected Resource(s)

* prefect_variable

### Terraform Configuration Files

```tf
resource "prefect_variable" "some_variable" {
  name = "some_variable_name"
  value = jsonencode([])
}
```

### Expected Behavior

After creating the variable, I expected the variable's value to be a JSON.

After modifying the value from `"[]"` to `["some_value"]` (notice the absence of surrounding double-quotes) in the UI and trying to re-apply the terraform plan: I would expect it to convert the JSON to a String so I can decode it and use it.

### Actual Behavior

After creating the variable, the value is a string `"[]"`

Applying the plan results in an error:
`Could not get Variable, unexpected error: failed to decode response: json: cannot unmarshal array into Go struct field Variable.value of type string`

Even when adding an `ignore_changes` lifecycle rule, the problem persists. (Which makes sense. It's still a data source).

### Steps to Reproduce

1. `terraform apply`
2. In the Cloud UI, change the value from `"[]"` to `["some_value"]`
3. `terraform apply`

### Important Factoids

N/A

### References

N/A

**Comment by parkedwards** [comment]
thanks for raising @felixpelletier - let us take a look at this

**Comment by mitchnielsen** [comment]
Alrighty, looks like I was able to replicate this:

First, I created a variable in the Prefect UI and gave it a value of `["some_value"]`.

Then I created the Terraform config to read that variable as a datasource:

```terraform
```

**Comment by mitchnielsen** [comment]
So what's the fix? We'll want to change the underlying Terraform Type of Variable values from `string` to `JSON`. This is due to changes made fairly recently to Variables in the API to make them JSON objects instead of just strings (example: https://github.com/PrefectHQ/prefect/pull/13543).

We'll go ahead and do that in the provider and publish that change as soon as we can. In the meantime, you should be able to work with plain strings just fine. Using JSON objects (arrays, maps, etc) will be supported once we push that update.

Thanks again @felixpelletier!

**Comment by mitchnielsen** [comment]
Hi @felixpelletier, we've released [v2.5.0](https://github.com/PrefectHQ/terraform-provider-prefect/releases/tag/v2.5.0) which includes a change to the `variable` value type to a dynamic Terraform type, which supports more types than just a string. The specific types supported are listed in the docs, and we're definitely open to supporting more types in the future depending on use cases.

Thanks again for reporting this 👍🏼
